dsc/db/models.py

```diff
@@ -0,0 +1,145 @@
+from enum import StrEnum
+from typing import TypedDict, Unpack
+
+from pynamodb.attributes import (
+    JSONAttribute,
+    NumberAttribute,
+    UnicodeAttribute,
+    UTCDateTimeAttribute,
+)
+from pynamodb.exceptions import PutError
+from pynamodb.models import Model
+
+from dsc.db.exceptions import ItemSubmissionExistsError
+
+
+class ItemSubmissionStatus(StrEnum):
+    RECONCILE_SUCCESS = "reconcile_success"
+    RECONCILE_FAILED = "reconcile_failed"
+    SUBMIT_SUCCESS = "submit_success"
+    SUBMIT_FAILED = "submit_failed"
+    SUBMIT_MAX_RETRIES_REACHED = "submit_max_retries_reached"
+    INGEST_SUCCESS = "ingest_success"
+    INGEST_FAILED = "ingest_failed"
+    INGEST_UNKNOWN = "ingest_unknown"
+    INGEST_MAX_RETRIES_REACHED = "ingest_max_retries_reached"
```

Comment:
+1 to defining the allowed statuses here.  Do you have plans to use this elsewhere in code?  Perhaps in the business logic when a particular status will get set?

Even if this commit / PR doesn't do that, I think this enum sets us up nicely for that.